ha/core/event_manager/model/action_event.py

```diff
@@ -29,20 +28,19 @@ def __init__(self, healthevent: HealthEvent):
         """
         Init method.
         """
-        self.event = Event()
         payload = {
-            HEALTH_ATTRIBUTES.SOURCE.value : healthevent.source,
-            HEALTH_ATTRIBUTES.CLUSTER_ID.value : healthevent.cluster_id,
-            HEALTH_ATTRIBUTES.SITE_ID.value : healthevent.site_id,
-            HEALTH_ATTRIBUTES.RACK_ID.value : healthevent.rack_id,
-            HEALTH_ATTRIBUTES.STORAGESET_ID.value : healthevent.storageset_id,
-            HEALTH_ATTRIBUTES.NODE_ID.value : healthevent.node_id,
-            HEALTH_ATTRIBUTES.RESOURCE_TYPE.value : healthevent.resource_type,
-            HEALTH_ATTRIBUTES.RESOURCE_ID.value : healthevent.resource_id,
-            HEALTH_ATTRIBUTES.RESOURCE_STATUS.value : healthevent.event_type,
-            HEALTH_ATTRIBUTES.SPECIFIC_INFO.value : healthevent.specific_info,
+            HealthAttr.SOURCE: healthevent.source,
+            HealthAttr.CLUSTER_ID: healthevent.cluster_id,
+            HealthAttr.SITE_ID: healthevent.site_id,
+            HealthAttr.RACK_ID: healthevent.rack_id,
+            HealthAttr.STORAGESET_ID: healthevent.storageset_id,
+            HealthAttr.NODE_ID: healthevent.node_id,
+            HealthAttr.RESOURCE_TYPE: healthevent.resource_type,
+            HealthAttr.RESOURCE_ID: healthevent.resource_id,
+            HealthAttr.RESOURCE_STATUS: healthevent.event_type,
         }
-        self.event.set_payload(payload)
+        self.event = HEvent(**payload)
+        self.event.set_specific_info({HealthAttr.SPECIFIC_INFO: healthevent.specific_info})
```

Comment:
Do we need this key as well, HealthAttr.SPECIFIC_INFO?

Reply:
No as set_specific_info function handle this : 

    def set_payload_attr(self, key, val):
        super().set(f'payload>{key}', val)

def set_specific_info(self, spec_info: dict):
        """Set payload attribute of key/value pairs specific to the resource type."""
        specific_info = KvPayload(spec_info)
        for key in specific_info.get_keys():
            super().set_payload_attr(f'specific_info>{key}', specific_info.get(key))

Reply:
in Utils it handle that : 
 def set_payload(self, payload: KvPayload):
        for key in payload.get_keys():
            super().set(f'payload>{key}', payload.get(key))


    def set_specific_info(self, spec_info: dict):
        """Set payload attribute of key/value pairs specific to the resource type."""
        specific_info = KvPayload(spec_info)
        for key in specific_info.get_keys():
            super().set_payload_attr(f'specific_info>{key}', specific_info.get(key))